auth0/src/main/java/com/auth0/android/authentication/storage/CredentialsManager.java

```diff
@@ -140,6 +164,37 @@ public void onFailure(@NonNull AuthenticationException error) {
                 callback.onFailure(new CredentialsManagerException("An error occurred while trying to use the Refresh Token to renew the Credentials.", error));
             }
         });
+
+    }
+
+    private boolean hasScopeChanged(@NonNull String storedScope, @Nullable String requiredScope) {
+        if (requiredScope == null) {
+            return false;
+        }
+        String[] stored = storedScope.split(" ");
+        Arrays.sort(stored);
+        String[] required = requiredScope.split(" ");
+        Arrays.sort(required);
+        return stored != required;
+    }
+
+    private boolean willExpire(long expiresAt, long minTtl) {
+        long nextClock = getCurrentTimeInMillis() + minTtl * 1000;
+        return expiresAt <= nextClock;
+    }
+
+    private long calculateExpiresAt(@NonNull Credentials credentials) {
+        long expiresAt = credentials.getExpiresAt().getTime();
+
+        if (credentials.getIdToken() != null) {
```

Comment:
The original use case that motivated this was about sending an Access Token (with enough lifetime left) to the backend. Shouldn't just the AT expire time be used here, instead of involving the ID Token as well?  See https://github.com/auth0/Auth0.swift/blob/master/Auth0/CredentialsManager.swift#L205.

`getCredentials()` uses this method to calculate the value that is then passed to `willExpire()`.

Reply:
I based my solution on your implementation but I'm keeping compatibility with the feature requested a while ago [here](https://github.com/auth0/Auth0.Android/issues/253). The TTL will consider both expirations and pick the most short one when comparing.

Reply:
In Auth0.swift there's 2 methods: `hasExpired` (that considers the ID Token) and `willExpire` (that doesn't). This way, if the ID Token has expired it will still renew the credentials.
See https://github.com/auth0/Auth0.swift/blob/master/Auth0/CredentialsManager.swift#L175

Reply:
There's no need to have two methods. If `willExpire` is passed 0 is the same as calculating a `hasExpired`, so both conditions can be checked with a single method.

Reply:
The point to have two methods is to check ID Token in one and not in the other.